frontend/src/screens/CartScreen.js

```diff
@@ -0,0 +1,99 @@
+        dispatch(removeFromCart(id))
+    }
+
+    const checkoutHandler = () => {
+       history.push('/login?redirect=shipping')
+    }
+
+    return (
+        <Row>
+            <Col md={8}>
+                <h1>Shopping Cart</h1>
+
+                {cartItems.length === 0 ? (
+                    <Message>Your cart is empty</Message>
+                ) : (
+                        <ListGroup variant='flush'>
+                            {cartItems.map((item) => (
+                                <ListGroup.Item key={item.product} >
+                                    <Row>
+                                        <Col md={2}>
+                                            <Image src={item.image} alt={item.name} fluid rounded />
+                                        </Col>
+                                        <Col md={3}>
+                                            <Link to={`/product/${item.product}`}>{item.name}</Link>
+                                        </Col>
+                                        <Col md={2}>${item.price}</Col>
+                                        <Col md={2}>
+                                            <Form.Control as='select' value={item.qty}
+                                                onChange={(e) =>
+                                                    dispatch(addToCart(item.product, Number(e.target.value)))
+                                                }>
+                                            {[...Array(item.countInStuck).keys()].map((x) => (
+                                                <option key={x + 1} value={x + 1}>
+                                                    {x + 1}
+                                                </option>
+                                            ))}
+                                            </Form.Control>
+                                        </Col>
+                                        <Col md={2}>
+                                            <Button type='button' variant='light' onClick={()=>
+                                        removeFromCartHandler(item.product)}>
+                                            <i className='fas fa-trash'>remove</i>
+                                        </Button>
+                                        </Col>
+                                    </Row>
+                                </ListGroup.Item>
+                            ))}
+                        </ListGroup>
+                    )}
+
+                    <Col md={4}>
+                        <Card>
+                            <ListGroup variant='flush'>
+                                <ListGroup.Item>
+                                    <h2>
+                                    Total Items ({cartItems.reduce((acc,item) => acc + item.qty,0)})
+                                    </h2>
+                                    Total Price({cartItems.reduce((acc,item) => acc + item.qty * item.price , 0).toFixed(2)})
```

Comment:
it's recommended to make calculations outside the JSX in a new variable and use the variable in the JSX

`const totalPrice = cartItems.reduce((acc,item) => acc + item.qty * item.price , 0).toFixed(2);`
`return (`
`some JSX`
`  Total Price({totalPrice})`
`some JSX`
`)`

Reply:
Okay I'll edit it   U+1F44D